en/2/11-interactingcontracts2.md

```diff
@@ -3,146 +3,293 @@ title: Using an Interface
+            &self, 
+            zombie_id: usize,
+            kitty_id: usize,
+          ) {
+            let crypto_kitties_sc_address = self.crypto_kitties_sc_address().get();
+              self.kitty_proxy(crypto_kitties_sc_address)
+                  .get_kitty(kitty_id)
+                  .async_call()
+                  .with_callback(self.callbacks().get_kitty_callback(zombie_id))
+                  .call_and_exit();
+          }
+          #[proxy]
+          fn kitty_proxy(&self, to: ManagedAddress) -> crypto_kitties_proxy::Proxy<Self::Api>;
       }
 ---
 
 Continuing our previous example with `NumberInterface`, once we've defined the interface as:
 
+A part of our contract that used this proxy would look like this:
+
 ```
-contract NumberInterface {
-  function getNum(address _myAddress) public view returns (uint);
-}
-```
+    #[endpoint]
+    fn call_adder_and_save_sum(&self) {
+      let caller = self.blockchain().get_caller();
+      let adder_contract_address = self.adder_contract_address().get();
+      self.adder_proxy(adder_contract_address)
+          .add(Biguint::from(2u32), Biguint::from(2u32))
+          .async_call()
+          .with_callback(self.callbacks().add_callback(&caller))
+          .call_and_exit();
+    }
 
-We can use it in a contract as follows:
+    #[callback]
+    fn add_callback(
+      &self, 
+      #[call_result] result: ManagedAsyncCallResult<BigUint>,
+      caller: &ManagedAddress
+    ) {
+        match result {
+            ManagedAsyncCallResult::Ok(sum) => {
+                self.sum(caller).set(sum);
+            },
+            ManagedAsyncCallResult::Err(_) => {
+                // this can only fail if the adder contract address is invalid
+                // nothing to revert in case of error
+            },
+        }
+    }
+
+    #[view(getSum)]
+    #[storage_mapper("sum")]
+    fn adder_contract_address(&self) -> SingleValueMapper<ManagedAddress>;
+
+    #[view(getSum)]
+    #[storage_mapper("sum")]
+    fn sum(&self, caller: &ManagedAddress) -> SingleValueMapper<BigUint>;
 
 ```
-contract MyContract {
-  address NumberInterfaceAddress = 0xab38... 
-  // ^ The address of the FavoriteNumber contract on Ethereum
-  NumberInterface numberContract = NumberInterface(NumberInterfaceAddress);
-  // Now `numberContract` is pointing to the other contract
-
-  function someFunction() public {
-    // Now we can call `getNum` from that contract:
-    uint num = numberContract.getNum(msg.sender);
-    // ...and do something with `num` here
-  }
-}
-```
 
-In this way, your contract can interact with any other contract on the Ethereum blockchain, as long they expose those functions as `public` or `external`.
+When calling another contract depending if both contracts are on same shard or on diferent shards we can make a direct call or a asynchronus call to one of their endpoints. In this context we can assume that the contracts are on diferent shards, reason why we used `async_call`.
```

Comment:
"if both contracts are in the same shard or not" - to shorten the phrase a bit.
"or an asynchronous call"